evap/staff/tests/test_tools.py

```diff
@@ -217,6 +220,70 @@ def test_do_nothing_if_test_run(self):
         self.assertEqual(len(messages), 4)
 
 
+class RemoveParticipationDueToInactivityTest(TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        cls.user = baker.make(UserProfile)
+        cls.evaluation = baker.make(
+            Evaluation,
+            state=Evaluation.State.PUBLISHED,
+            vote_start_datetime=datetime.today() - timedelta(days=200),
+            vote_end_date=(datetime.today() - timedelta(days=180)).date(),
+            participants=[cls.user],
+        )
+        cls.evaluation.course.semester.archive()
+
+    @override_settings(PARTICIPATION_DELETION_AFTER_INACTIVE_TIME=timedelta(6 * 30))
+    def test_remove_user_due_to_inactivity(self):
+        self.assertTrue(self.user.evaluations_participating_in.exists())
+
+        messages = remove_inactive_participations(self.user)
+
+        self.assertFalse(self.user.evaluations_participating_in.exists())
+        self.assertTrue(self.user.can_be_marked_inactive_by_manager)
+        self.assertEqual(messages, [f"1 participation of {self.user.full_name} was removed due to inactivity."])
+
+        messages = remove_inactive_participations(self.user)
+
+        self.assertEqual(messages, [])
+
+    @patch("evap.evaluation.models.UserProfile.is_active", True)
+    @patch("evap.evaluation.models.UserProfile.can_be_marked_inactive_by_manager", True)
+    @override_settings(PARTICIPATION_DELETION_AFTER_INACTIVE_TIME=timedelta(360))
+    def test_do_not_remove_user_due_to_inactivity_with_recently_archived_evaluation(self):
+        self.assertTrue(self.user.evaluations_participating_in.exists())
+
+        messages = remove_inactive_participations(self.user)
+
+        self.assertTrue(self.user.evaluations_participating_in.exists())
```

Comment:
We can be more explicit here (and below):
```suggestion
        self.assertEqual(list(self.user.evaluations_participating_in.all()), [self.evaluation])
```

Reply:
Done.